elevate-cpanel

```diff
@@ -1982,6 +1982,15 @@ EOS
         my $errors = Cpanel::SafeRun::Errors::saferunonlyerrors(qw{/usr/bin/yum makecache});
         if ( $errors =~ m/\S/ms ) {
 
+        my $error_msg = <<~'EOS';
+        '/usr/bin/yum makecache' failed to return cleanly.  This could be due to a temporary mirror problem or it could be indicative of a larger issue such as a broken repository.  Since this script relies heavily on yum, this will need to be addressed before upgrading.
+
+        You may want to consider reaching out to cPanel Support for assistance:
```

Comment:
I'm concerned with this advice. This advice could be supplied everywhere in theory. why here?

Reply:
We already recommend this in a few spots:

```
# git grep 'how-to-open-a-technical-support-ticket' lib/ script/
lib/Elevate/Blockers/Grub2.pm:        https://docs.cpanel.net/knowledge-base/technical-support-services/how-to-open-a-technical-support-ticket/
lib/Elevate/Blockers/Grub2.pm:    https://docs.cpanel.net/knowledge-base/technical-support-services/how-to-open-a-technical-support-ticket/
lib/Elevate/Blockers/Grub2.pm:        https://docs.cpanel.net/knowledge-base/technical-support-services/how-to-open-a-technical-support-ticket/
lib/Elevate/Blockers/MountPoints.pm:        https://docs.cpanel.net/knowledge-base/technical-support-services/how-to-open-a-technical-support-ticket/
lib/Elevate/Components/CCS.pm:        https://docs.cpanel.net/knowledge-base/technical-support-services/how-to-open-a-technical-support-ticket/
```

I thought it made sense to do it in this instance because there are lots of different ways that `yum makecache` can fail and there is not a good automated way to determine why it failed due to the number of errors that could be emitted from it.  It also made sense to me here because support deals with yum issues, including makecache failing, quite frequently due to EA4 among other things so they should be well equiped to handle issues that come in due to this.

Reply:
What @cPholloway was getting at was more or less my take.

If we wanted to extend some method to be more useful we would probably need to identify something like the top x known errors after parsing through tickets to identify what common errors were found - then implementing some sort of regex to trigger whatever text advise etc we would want to display if one of those common errors was identified.

Past that we could just look for core patterns like:

- /Resolving Dependencies/&&/Requires:\s/ - could obviously show a dependency issue taking place and recommend filing a ticket
- /Error: requested datatype/ - could signify a bad repo is sitting stale in /etc/yum.repos.d/*.repo

This just seemed the simplest thing for the surface area.